packages/nimble-components/src/mapping/base/index.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    attr,
+    booleanConverter,
+    nullableNumberConverter
+} from '@microsoft/fast-element';
+import { FoundationElement } from '@microsoft/fast-foundation';
+import type { MappingConfig } from '../../table-column/enum-base';
+
+/**
+ * An element to be given as content to a nimble-table-column-mapping or nimble-table-column-icon.
+ * This is a base class from which other specific mapping types (e.g. MappingIcon, MappingText) are derived.
+ */
+export abstract class Mapping extends FoundationElement {
+    @attr({ mode: 'fromView' })
+    public key?: string | number | boolean;
+
+    @attr({ attribute: 'default-mapping', mode: 'boolean' })
+    public defaultMapping = false;
+
+    public abstract getConvertedKeyMapping(
+        keyType: 'string' | 'number' | 'boolean'
```

Comment:
Would it work to use `MappingKeyType` in this file and derived classes instead of `'string' | 'number' | 'boolean'`? Even if not, I'd vote to define a type for this to make it easier to update if we ever decide to support more key types.

Reply:
Yep, this was an oversight.